service/src/main/java/org/ehrbase/plugin/PluginAspect.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2022. vitasystems GmbH and Hannover Medical School.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.ehrbase.plugin;
+
+import com.nedap.archie.rm.composition.Composition;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.ehrbase.api.exception.InternalServerException;
+import org.ehrbase.plugin.dto.CompositionWithEhrId;
+import org.ehrbase.plugin.extensionpoints.CompositionExtensionPointInterface;
+import org.springframework.beans.factory.ListableBeanFactory;
+import org.springframework.core.annotation.AnnotationAwareOrderComparator;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * @author Stefan Spiska
+ */
+@Component
+@Aspect
+public class PluginAspect {
```

Comment:
@ConditionalOnProperty(prefix = PLUGIN_MANAGER_PREFIX, name = "enable", havingValue = "true")
should be added here, since we do not need this aspect in the invocation chain if we do not use plugins